Xamarin.Forms.Platform.Android/CollectionView/StructuredItemsViewAdapter.cs

```diff
@@ -139,7 +139,12 @@ protected RecyclerView.ViewHolder CreateHeaderFooterViewHolder(object content, D
 
 			if (content is View formsView)
 			{
-				return SimpleViewHolder.FromFormsView(formsView, context);
+				var viewHolder = SimpleViewHolder.FromFormsView(formsView, context);
+
+				// Propagate the binding context from the ItemsView to the header/footer
+				viewHolder.View.BindingContext = ItemsView.BindingContext;
```

Comment:
Shoudn't we use SetInheritedBindingContext ?

Reply:
What does SetInheritedBindingContext do?

Reply:
Does a couple of more checks if there's a BC on the View , this will actually how other places do it..

https://github.com/xamarin/Xamarin.Forms/blob/master/Xamarin.Forms.Core/BindableObject.cs#L185

Reply:
We're both wrong :)

The problem is that the header/footer views aren't being added as children of the CollectionView, so they aren't getting the BindingContext from the CollectionView, or the FlowDirection, or Visual. So AddLogicalChild is what we're looking for.

Reply:
Oh didn't remember that one. Nice catch!